src/ATen/native/xpu/Activation.cpp

```diff
@@ -396,4 +397,48 @@ Tensor XPUNativeFunctions::hardswish_backward(
   return iter.output();
 }
 
+static void check_even_dimension(const Tensor& input, int64_t dim) {
+  auto wrap_dim = maybe_wrap_dim(dim, input.dim());
+  const int64_t nln = input.size(wrap_dim);
+  TORCH_CHECK(
+      nln % 2 == 0,
+      "Halving dimension must be even, but dimension",
+      wrap_dim,
+      " is size ",
+      nln);
+}
+
+Tensor& XPUNativeFunctions::glu_out(
```

Comment:
Please move the op to `src/ATen/native/xpu/GatedLinearUnit.cpp` to align with stock PyTorch.